evalml/data_checks/natural_language_nan_data_check.py

```diff
@@ -0,0 +1,61 @@
+from evalml.data_checks import DataCheck, DataCheckError, DataCheckMessageCode
+from evalml.utils.woodwork_utils import (
+    _convert_woodwork_types_wrapper,
+    infer_feature_types
+)
+
+error_contains_nan = "Input natural language column(s) ({}) contains NaN values. Please impute NaN values or drop these rows or columns."
+
+
+class NaturalLanguageNaNDataCheck(DataCheck):
+    """Checks if natural language columns contain NaN values."""
+
+    def __init__(self):
+        """Checks each column in the input for natural language features and will issue an error if NaN values are present.
+        """
+
+    def validate(self, X, y=None):
+        """Checks if any natural language columns contain NaN values.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame, np.ndarray): Features.
+            y (ww.DataColumn, pd.Series, np.ndarray): Ignored.  Defaults to None.
+
+        Returns:
+            dict: dict with a DataCheckError if NaN values are present in natural language columns.
+
+        Example:
+            >>> import pandas as pd
+            >>> import woodwork as ww
+            >>> import numpy as np
+            >>> data = pd.DataFrame()
+            >>> data['A'] = [None, "string_that_is_long_enough_for_natural_language"]
+            >>> data['B'] = ['string_that_is_long_enough_for_natural_language', 'string_that_is_long_enough_for_natural_language']
```

Comment:
Hehe nit-pick / I don't think we did a good job of doing this before but maybe just explicitly set some cols as nat lang cols? Just thinking if WW ever changes their type inference, this might break and no longer be considered nat lang.

Reply:
good call - I'll add it to the docstring and the explicit WW tests!